vault/resource_pki_secret_backend_config_cluster.go

```diff
@@ -0,0 +1,161 @@
+				id := d.Id()
+				if id == "" {
+					return nil, fmt.Errorf("no path set for import, id=%q", id)
+				}
+
+				parts := strings.Split(util.NormalizeMountPath(id), "/")
+				if err := d.Set("backend", parts[0]); err != nil {
+					return nil, err
+				}
+
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			consts.FieldBackend: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Full path where PKI backend is mounted.",
+			},
+			consts.FieldPath: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to the cluster's API mount path.",
+			},
+			consts.FieldAIAPath: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Path to the cluster's AIA distribution point.",
+			},
+		},
+	}
+}
+
+func pkiSecretBackendConfigClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	path := fmt.Sprintf("%s/config/cluster", backend)
+
+	resp, err := client.Logical().ReadWithContext(ctx, path)
+	if err != nil {
+		return diag.Errorf("error reading cluster config at %s, err=%s", path, err)
+	}
+
+	if resp == nil {
+		return diag.Errorf("no cluster config found at path %s", path)
+	}
+
+	d.SetId(path)
+
+	return pkiSecretBackendConfigClusterUpdate(ctx, d, meta)
+}
+
+func pkiSecretBackendConfigClusterUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	path := d.Id()
+
+	fields := []string{
+		consts.FieldPath,
+		consts.FieldAIAPath,
+	}
+
+	var patchRequired bool
+	data := map[string]interface{}{}
+	for _, k := range fields {
+		if d.HasChange(k) {
+			data[k] = d.Get(k)
+			patchRequired = true
+		}
+	}
+
+	if patchRequired {
+		_, err := client.Logical().WriteWithContext(ctx, path, data)
+		if err != nil {
+			return diag.Errorf("error writing data to %q, err=%s", path, err)
+		}
+	}
+
+	return pkiSecretBackendConfigClusterRead(ctx, d, meta)
+}
+
+func pkiSecretBackendConfigClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, e := provider.GetClient(d, meta)
+	if e != nil {
+		return diag.FromErr(e)
+	}
+
+	path := d.Id()
+
+	if path == "" {
+		return diag.Errorf("no path set, id=%q", d.Id())
+	}
+
+	log.Printf("[DEBUG] Reading %s from Vault", path)
+	resp, err := client.Logical().ReadWithContext(ctx, path)
+	if err != nil {
+		return diag.Errorf("error reading from Vault: %s", err)
+	}
+
+	if resp == nil {
+		backend := d.Get(consts.FieldBackend).(string)
+		path := fmt.Sprintf("%s/config/cluster", backend)
+		d.SetId(path)
```

Comment:
We must unset the ID here otherwise TF will not know that this resource does not exist.
```suggestion
		d.SetId("")
```

Reply:
Right, thanks.